supplier_app/forms.py

```diff
@@ -45,7 +47,40 @@ def is_valid(self):
         return False
 
 
-class AddressCreateForm(ModelForm):
+class BasePrefixCreateForm(ModelForm):
+    def __init__(self, data=None, files=None, *args, **kwargs):
+
+        data_query_dict = None
+        files_query_dict = None
+        if not self.prefix:
+            raise ValueError("You need to specify the prefix attribute")
+        if data or files:
+            data_query_dict = QueryDict('', mutable=True)
+            files_query_dict = QueryDict('', mutable=True)
+            data = data or {}
+            files = files or {}
+            data_query_dict.update(
+                {
+                    data_key: data_value
+                    for data_key, data_value in data.items()
+                    if data_key.startswith(self.prefix)
+                }
+            )
+            files_query_dict.update(
+                {
+                    file_key: file_value
+                    for file_key, file_value in files.items()
+                    if file_key.startswith(self.prefix)
+                }
+            )
```

Comment:
seems like data and files are making the same steps, could be unified on a function?

Reply:
Yes you are right, I will try to implement that

Reply:
Resolved